### ISSUE TITLE
Language package status as of v1.3.2

### ISSUE THREAD

**Issue by TakWolf** [aseprite/languages#26]
The following information should be updated to README.md

--------------------------------------------

## Language package status

As of v1.3.2

| Language | Repository | Progress | Status |
|---|---|---:|---:|
| Chinese (Simplified) | [aseprite-quest/aseprite-language-chinese-simplified](https://github.com/aseprite-quest/aseprite-language-chinese-simplified) | 1613 / 1613 (100.00% 🚩) | Positive |
| Chinese (Traditional) | [5idereal/Aseprite-Traditional-Chinese-Translation](https://github.com/5idereal/Aseprite-Traditional-Chinese-Translation) | 1613 / 1613 (100.00% 🚩) | Positive |
| Russian | [lufog/aseprite-language-russian](https://github.com/lufog/aseprite-language-russian) | 1613 / 1613 (100.00% 🚩) | Positive |
| Italian | [FabianoIlCapo/aseprite_italian](https://github.com/FabianoIlCapo/aseprite_italian) | 1593 / 1613 (98.76% 🚧) | Positive |
| French | [boubl/Aseprite-French-Translation](https://github.com/boubl/Aseprite-French-Translation) | 1613 / 1613 (100.00% 🚩) | Positive |
| Japanese | [aseprite-quest/aseprite-language-japanese](https://github.com/aseprite-quest/aseprite-language-japanese) | 1194 / 1613 (74.02% 🚧) | Looking for Maintainer |
| Korean | [ImBada/Aseprite-Korean](https://github.com/ImBada/Aseprite-Korean) | 1193 / 1613 (73.96% 🚧) | Inactive |
| German | [dotheflopboy/Aseprite-German-Translation](https://github.com/dotheflopboy/Aseprite-German-Translation) | 1154 / 1613 (71.54% 🚧) | Inactive |
| Spanish | [raxdraws/aseprite-spanish](https://github.com/raxdraws/aseprite-spanish) | 1187 / 1613 (73.59% 🚧) | Inactive |
| Ukrainian | [Steenuga/aseprite-language-ukrainian](https://github.com/Steenuga/aseprite-language-ukrainian) | 1283 / 1613 (79.54% 🚧) | Inactive |
| Polish | [PxSprite/aseprite-beta-pl](https://github.com/PxSprite/aseprite-beta-pl) | 1283 / 1613 (79.54% 🚧) | Inactive |
| Portuguese (Brazil) | [puddiCria/aseprite-pt-br](https://github.com/puddiCria/aseprite-pt-br) | 1239 / 1613 (76.81% 🚧) | Inactive |
| Maltese | [DimensionalSpace/Maltese-Aseprite](https://github.com/DimensionalSpace/Maltese-Aseprite) | 1539 / 1613 (95.41% 🚧) | Inactive |

## Other packages

The following packages is not the best or lack of maintenance.

- Chinese (Simplified)
    - [J-11/Aseprite-Simplified-Chinese](https://github.com/J-11/Aseprite-Simplified-Chinese) - [steamcommunity](https://steamcommunity.com/sharedfiles/filedetails/?id=1333477949)
- Chinese (Traditional)
    - [chongx1an/aseprite-TraditionalChineseExtension](https://github.com/chongx1an/aseprite-TraditionalChineseExtension)
- French
    - [farvardin/aseprite_french](https://github.com/farvardin/aseprite_french)
- Japanese
    - [日本語化ファイルのダウンロード](http://wikiwiki.jp/aseprite/?%C6%FC%CB%DC%B8%EC%B2%BD%A5%D5%A5%A1%A5%A4%A5%EB%A4%CE%A5%C0%A5%A6%A5%F3%A5%ED%A1%BC%A5%C9)
- Korean
    - [eternalworld.tistory.com](https://eternalworld.tistory.com/531)
- Portuguese
    - [Inky1003/aseprite-em-portugues](https://github.com/Inky1003/aseprite-em-portugues)
- Russian
    - [xMrVizzy/aseprite_russian](https://github.com/xMrVizzy/aseprite_russian)
- Spanish
    - [Latirus/aseprite-spanish](https://github.com/Latirus/aseprite-spanish) 
    - [Dolvem/aseprite-spanish-translation](https://github.com/Dolvem/aseprite-spanish-translation)

**Comment by dacap** [comment]
Hi @TakWolf, thanks a lot for the work you are doing with this, I'll update the README with your info, anyway I'm not sure how could we automatize this.

In some way the third-party extensions for languages was a good idea, but probably we could start integrating translations in some way (through some kind of service, e.g. #8?). Or probably offering tools to facility this/download languages (I saw that you have made a lot of progress with https://github.com/aseprite-quest/).

One possible idea would be to have "officially linked translations", so we can offer then by default in the combobox:
![Screenshot 2023-12-04 at 11 51 55](https://github.com/aseprite/languages/assets/39654/f33ab32f-d30a-41bf-b199-793427eab5bf)

And when a translation is selected the extension is downloaded automatically (?)

**Comment by TakWolf** [comment]
Integrating translations and official built-in download/update language package maybe good ideas.

Most existing third-party language packs are not continuously updated. They usually only create once and then become inactive. I have been looking for a simple way for everyone to easily submit contributions. (Godot use https://hosted.weblate.org/projects/godot-engine/godot-class-reference/ , But this seems too heavy for Aseprite)

I am still exploring. Currently, I have written a simple diff tool that can pull out the 'en.ini' and generate differences based on existing 'lang.ini' files, like: 

https://github.com/aseprite-quest/aseprite-unified-languages/blob/332d6decd5fe1b1a580f9ccdc5b2428360e55527/data/ko.ini#L5-L29

```ini
[advanced_mode]
title = 경고 - 주의사항
description = "고급 모드"를 활성화하게 됩니다.

[ani_controls]
# TODO # shortcut = Shortcut:
# TODO # right_click = Right-click: Show playback options

[anidir_combo]
# TODO # forward = Forward
# TODO # reverse = Reverse
# TODO # ping_pong = Ping-pong
# TODO # ping_pong_reverse = Ping-pong Reverse

[ask_for_color_profile]
title = Color Profile
sprite_with_profile = The sprite contains a color profile.
sprite_without_profile = The sprite doesn't contain a color profile.

[statusbar_tips]
all_layers_are_locked = 선택된 모든 레이어가 잠겨있음
layer_locked = Layer '{0}' is locked
# TODO # disable_snap_grid = Disable Snap to Grid
# TODO # frame = Frame:
# TODO # current_frame = Current Frame
```

This way, anyone can easily replace '# TODO#' line to the correct translation. Combining git tools, it should be possible to avoid missing statements, and can also perform automatic statistics. If there is a noun dictionary, it should also ensure accuracy and consistency. On this basis, if integrated translation is used, it may improve language pack active.

This method is currently not perfect. If the key has not changed but the value text has changed, it cannot be alert.


Currently, https://github.com/aseprite-quest/aseprite-unified-languages can automatically pulling, statistics and packaging. At present, translations are all pulled from third-party projects, but if use integrated translation, can directly submit contributions to `data/*.ini`.
This should be able to provide some reference.

**Comment by dacap** [comment]
I see. In the past (a couple of times) I provided some git diff between versions (e.g. `git diff v1.2.20..v1.2.21`, etc. but it's quite hard to follow the changes, and doing all the paperwork just with tools.

I'm giving a try to Weblate (nice a alternative to Transifex #8 👍 ), I think we need some centralized way to organize the translations and keep then updated (and indicate when a string needs "re-translation"). And we can afford some medium plan for this.

One thing that might make things complex is that some specific translations need/required/or would prefer other fonts, we might need a way to indicate this too.

**Comment by TakWolf** [comment]
Solved by weblate

**Comment by Flaconia** [comment]
@dacap 
Hi, so at this very moment, the "official" translations are still the ones we've been using until now, or is the Weblate versions already working?
Asking this to know: how should a regular user download a localization at the moment?

**Comment by TakWolf** [comment]
@FabianoIlCapo 

My guess, starting from a later version [probability v1.3.3], translation will become built-in and no longer need the language extension packs.

Before the each version is released, translations will be synced from weblate to the [strings directory](https://github.com/aseprite/aseprite/tree/main/data/strings)

**Comment by dacap** [comment]
Hi there @FabianoIlCapo! As @TakWolf said we'll try to include the official translations with the next release (at least including the ones that are almost complete). For older versions (v1.3.2 or older) the extensions will be still needed.

There are two main issues that need some thought yet:

1. What happens if one user has an extension + the new official translation. I think we should include in the combobox of *Edit > Preferences > General > Language* the list of extensions translations and official translations (and a way to differentiate if the user chose the official one or a third-party one). Or probably we should detect what is the newest translation.
2. Should we offer a way to update translations from Weblate/the github repo? Probably we should only offer updated translations included with each new release and forget about auto-updating strings-only (which adds more complexity).

**Comment by Flaconia** [comment]
Thank you both for the info!

1. To simplify this dilemma, wouldn't it make sense to have the user just select the language they want and if the extensions come with that language included, they're set to it, otherwise said extensions just keep their original language?
Am I understanding it correctly, thinking that if we want this behaviour to be possible, the extensions and their translations will need to be managed officially as well?

2. Would this mean that for every new Aseprite release, the translations come "as they are" and cannot be updated until a new version of Aseprite comes out? (or I guess manually by overwriting the .ini?)

**Comment by dacap** [comment]
1. That's right, the simplest version (and as it works right now) all languages installed through extensions have more priority from the official ones, we can keep this behavior (and probably we should keep it?) even when the extension is outdated, although this might generate a confusing situation: where the extension translation is outdated and the official one isn't.

2. I'm thinking in possible ways to update translations from the GitHub repository automatically, but sometimes the translation might depend on the code (e.g. an old feature is replaced with a new implementation + new translated strings), so the new package doesn't work for old versions of Aseprite. In the best case the new release + new translations must go hand by hand (we could offer some way to offer updates for strings only, but it must be only for existent strings, the update process is a little tricky in this situation).

**Comment by TakWolf** [comment]
Note that the new translation format has been changed, (No more multiple lines: <<<END).
Therefore, the old language pack may not be compatible with the new language pack.

**Comment by dacap** [comment]
> Note that the new translation format has been changed, (No more multiple lines: <<<END). Therefore, the old language pack may not be compatible with the new language pack.

Actually the old translation files (the old multiple lines <<<END format) will be supported, so the problem is that new language files (the ones with \n in weblate) will not work for old versions of Aseprite.
